unittests/test_autodetect.py

```diff
@@ -65,6 +65,20 @@ def remote_exec_ctx(make_exec_ctx, temp_topo):
     yield ctx
 
 
+@pytest.fixture
+def remote_custom_exec_ctx(make_exec_ctx, temp_topo):
+    if test_util.USER_CONFIG_FILE is None:
+        pytest.skip('no user configuration file supplied')
+
+    ctx = make_exec_ctx(test_util.USER_CONFIG_FILE,
+                        test_util.USER_SYSTEM,
+                        {'general/remote_detect': True,
+                         'general/remote_command': [
+                             'echo \'{"dummy": "value"}\' > topo.json',
+                         ]})
+    yield ctx
```

Comment:
Just extend the existing `remote_exec_ctx` to accept additional configuration options.